Fix broken PNG due to incorrect chunk type detection #1928

creatissimo opened the pull request:
In order to fix https://github.com/mpdf/mpdf/issues/1928 I've refactored the ImageProcessor -> processPng function and replaced strpos function by getting chunk types from chunk bytes 5-8.
I've refactored all places and not only the gAMA section which was mentioned in the issue

finwe commented:
Thanks for the PR! Other than the failing CS check, this looks good. Please fix the coding standard, squash the commits to one and end the commit message to `(Closes #1928)`, I'll then be happy to merge.

creatissimo commented:
Thx for your feedback. I've fixed the CS issues and squashed the commits with the commit message ending with `(Closes #1928)`. Please let me know if I can be of further assistance
